internal/publish/publish.go

```diff
@@ -265,6 +267,28 @@ func (p *defaultPublisher) createDeploymentRecord(
 	return p.writeDeploymentRecord()
 }
 
+// Does the target exist and the user has permissions to it?
+func (p *defaultPublisher) targetPreflightChecks(client connect.APIClient, contentID types.ContentID) error {
```

Comment:
I noticed that because the error is retrieved from `ContentDetails` which does a `GET` call and these errors have codes plugged in, but nothing else, the error we write to the deployment record `message` is largely un-helpful. The message is just what the API call error has in `message`

Currently that is the error we show for the "Last Deployment Failed" message in the sidebar. So while I get a great message when the piece of content cannot be found in the notification window the sidebar error is quite generic:

![CleanShot 2024-10-02 at 13 00 20@2x](https://github.com/user-attachments/assets/c3673288-21f9-4717-9dec-23daa0f0ba87)

Not sure if we should consider that part of this work, but I'd argue because its a one line addition it should be.

Reply:
Maybe improvements to this messaging be related to this other issue #2206?

Reply:
The methodology to set that message is the same, but they are different parts of the publishing process. Ideally we would solve one at a time.

Reply:
I would think if we created some kind of error message factory, which took in the original error (present either in the stream or in the content record file), then we can consistently produce the enhanced message.